controllers/mdutil/util_test.go

```diff
@@ -122,6 +122,12 @@ func TestEqualIgnoreHash(t *testing.T) {
 			generateMachineTemplateSpec("foo", map[string]string{}, map[string]string{DefaultMachineDeploymentUniqueLabelKey: "value-2", "something": "else"}),
 			true,
 		},
+		{
+			"Same spec, the former doesn't have machine-template-hash label",
+			generateMachineTemplateSpec("foo", map[string]string{}, map[string]string{"something": "else"}),
+			generateMachineTemplateSpec("foo", map[string]string{}, map[string]string{DefaultMachineDeploymentUniqueLabelKey: "value-2", "something": "else"}),
+			true,
+		},
```

Comment:
Should there also be a test case for when the api versions of objectReferences differ?

Reply:
Yep added both